src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/ConfigurationTests.cs

```diff
@@ -0,0 +1,173 @@
+                File.WriteAllText(Path.Combine(contentRootDirectory.FullName, "appsettings.json"), appSettingsContent);
+            }
+            if (level >= ConfigurationLevel.UserSettings)
+            {
+                // This is the settings.json file in the user profile directory.
+                string userSettingsContent = JsonSerializer.Serialize(UserSettingsContent);
+                File.WriteAllText(Path.Combine(userConfigDir.FullName, "settings.json"), userSettingsContent);
+            }
+            if (level >= ConfigurationLevel.SharedSettings)
+            {
+                // This is the settings.json file in the shared configuration directory that is visible
+                // to all users on the machine e.g. /etc/dotnet-monitor on Unix systems.
+                string sharedSettingsContent = JsonSerializer.Serialize(SharedSettingsContent);
+                File.WriteAllText(Path.Combine(sharedConfigDir.FullName, "settings.json"), sharedSettingsContent);
+            }
+            if (level >= ConfigurationLevel.SharedKeyPerFile)
+            {
+                // This is a key-per-file file in the shared configuration directory. This configuration
+                // is typically used when mounting secrets from a Docker volume.
+                File.WriteAllText(Path.Combine(sharedConfigDir.FullName, WebHostDefaults.ServerUrlsKey), nameof(ConfigurationLevel.SharedKeyPerFile));
+            }
+
+            // Create the initial host builder.
+            IHostBuilder builder = HostBuilderHelper.CreateHostBuilder(settings);
+
+            // Override the environment configurations to use predefined values so that the test host
+            // doesn't inadvertently provide unexpected values. Passing null replaces with an empty
+            // in-memory collection source.
+            builder.ReplaceAspnetEnvironment(level >= ConfigurationLevel.AspnetEnvironment ? AspnetEnvironmentVariables : null);
+            builder.ReplaceDotnetEnvironment(level >= ConfigurationLevel.DotnetEnvironment ? DotnetEnvironmentVariables : null);
+            builder.ReplaceMonitorEnvironment(level >= ConfigurationLevel.MonitorEnvironment ? MonitorEnvironmentVariables : null);
+
+            // Build the host and get the Urls property from configuration.
+            IHost host = builder.Build();
+            IConfiguration rootConfiguration = host.Services.GetRequiredService<IConfiguration>();
+            string configuredUrls = rootConfiguration[WebHostDefaults.ServerUrlsKey];
+
+            // Test that the value of the Urls property is the same as it was set
+            // for the level of configuration of the test.
+            if (level == ConfigurationLevel.None)
+            {
+                Assert.Null(configuredUrls);
+            }
+            else if (level == ConfigurationLevel.DotnetEnvironment)
+            {
+                // Not sure this was the intent that setting DOTNET_Urls via environment yields no
```

Comment:
@wiktork, not sure if you had thoughts on this. It feels like setting DOTNET_Urls should override whatever the `--urls` argument is. It's kind of in a weird state right now where DOTNET_Urls has lower precedence but ASPNETCORE_Urls has higher precedence.

Reply:
I was under the impression DOTNET_Urls already supersedes the command line parameter. At least the intent was that the command line parameter is secondary to environment variables since we wanted docker configurability through environment variables.

Reply:
I can make a separate change to fix that.

Reply:
Sounds good. I probably just tested ASPNET_Urls in the initial implementation.

Reply:
Fix in #1382